src/hotspot/share/gc/g1/g1CollectionSet.cpp

```diff
@@ -232,7 +232,11 @@ void G1CollectionSet::iterate_part_from(HeapRegionClosure* cl,
                                         size_t offset,
                                         size_t length,
                                         uint worker_id) const {
-  _g1h->par_iterate_regions_array_part_from(cl, hr_claimer, _collection_set_regions, offset, length, worker_id);
+  _g1h->par_iterate_regions_array(cl,
+                                  hr_claimer,
+                                  &_collection_set_regions[offset],
+                                  length,
```

Comment:
This should be `length - offset`, shouldn't it?

Reply:
```suggestion
                                  length - offset,
```

Untested :)

Reply:
I though so too at first, but seems it's not.
The passed in length in both G1CollectionSet::par_iterate/iterate_incremental_part_from are both the length of the "sub" array already. ( increment_length() is passed as length in iterate_incremental_part_from)
How do you think about it?

Reply:
You are right.